src/components/map/MapDirective.js

```diff
@@ -63,22 +64,75 @@ goog.require('ga_permalink');
 
             // Update permalink based on view states.
             var updatePermalink = function() {
-              var center = view.getCenter();
-              var zoom = view.getZoom();
-              // when the directive is instantiated the view may not
-              // be defined yet.
-              if (center && zoom !== undefined) {
-                var x = center[1].toFixed(2);
-                var y = center[0].toFixed(2);
-                gaPermalink.updateParams({X: x, Y: y, zoom: zoom});
+              // only update the permalink in 2d mode
+              if (!scope.ol3d || !scope.ol3d.getEnabled()) {
+                // remove 3d params
+                gaPermalink.deleteParam('lon');
+                gaPermalink.deleteParam('lat');
+                gaPermalink.deleteParam('elevation');
+                gaPermalink.deleteParam('heading');
+                gaPermalink.deleteParam('pitch');
+                var center = view.getCenter();
+                var zoom = view.getZoom();
+                // when the directive is instantiated the view may not
+                // be defined yet.
+                if (center && zoom !== undefined) {
+                  var x = center[1].toFixed(2);
+                  var y = center[0].toFixed(2);
+                  gaPermalink.updateParams({X: x, Y: y, zoom: zoom});
+                }
               }
             };
             view.on('propertychange', gaDebounce.debounce(updatePermalink,
                 1000, false));
-            updatePermalink();
```

Comment:
+1 for "We won't have X/Y/zoom in the PL after the page is loaded and the
user didn't yet interact with it" makes sense (and it was like this in
RE2...)

On Thu, Aug 27, 2015 at 8:14 AM, Gilbert Jeiziner <notifications@github.com>
wrote:

> In src/components/map/MapDirective.js
> <https://github.com/geoadmin/mf-geoadmin3/pull/2593#discussion_r38065129>:
>
> >                }
> >              };
> >              view.on('propertychange', gaDebounce.debounce(updatePermalink,
> >                  1000, false));
> > -            updatePermalink();
>
> The state is the default state.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2593/files#r38065129>.
>